test/functional/helpers.lua

```diff
@@ -319,6 +319,7 @@ end
 -- Dedent the given text and write it to the file name.
 local function write_file(name, text, dont_dedent)
   local file = io.open(name, 'w')
+  if file == nil then return false end
```

Comment:
Was this silently failing before? If not, it will now, which means we would have to explicitly check `eq(true, write_file())` everywhere.

Reply:
It was failing for `ex_cmds/write_spec.lua` because `os.remove` did not remove the files. `write_file` tried to write on a directory so `file:write` raised an error.